[AWSIC] Adds RoleSyncMode selection to `tctl plugins` tooling

tcsc opened the pull request:
Allows the user to specify a RolesSyncMode on the command line when installing or editing the
Identity Center integration.

Also prevents the user from specifying group import filters on installation when `roleSyncMode == NONE`, because allowing them will make an invalid plugin configuration.

Changelog: Added ability for user to select whether IC integration creates roles for all possible Account Assignments

r0mant reviewed (commented):
@tcsc Can you please include documentation in this PR as well.

r0mant reviewed (commented):
> Changelog: Added RolesSyncMode selection to tctl AWSIC plugin installer

@tcsc Also, this changelog entry will not make sense to anyone outside of IG team and certainly not users reading this who are the primary consumers of the changelog. Can you rephrase it to be more human readable?

tcsc commented:
Accompanying documentation is in #58345

r0mant reviewed (approved):
lgtm with some wording suggestions

tcsc committed:
[AWSIC] Adds RoleSyncMode selection to `tctl plugins install awsic`

Allows the user to specify a RolesSyncMode on the command line when installing or
editing the Identity Center integration. Also prevents the user from specifying
group import filters when rolesSyncMode == NONE, because allowing them will make
an invalid plugin configuration.

tcsc committed:
Merge branch 'master' into tcsc/tctl-set-role-sync-mode
